ocrd/ocrd/decorators/parameter_option.py

```diff
@@ -5,6 +5,7 @@
 
 
 def _handle_param_option(ctx, param, value):
+    # FIXME: use resolve_resource for preset file names
```

Comment:
> Also, we cannot defer decoding the param value (as JSON or path name) early in the decorator callback to some later time in the Processor constructor, because crucially, for the `--param-override` mechanism we need to combine with the parsed value of any `--param`.

We could take this as a reason to move the whole parameter_option handling from the command option decorator to the Processor constructor. Or, conversely, instantiate a throwaway processor (without arguments) just for `resolve_resource`.

Reply:
The latter is not possible from the option decorator (because the wrapped class is not known yet).

The former is still problematic, because `Processor.resolve_resource` cannot be used (it exits when no candidate can be resolved) and it must happen inside of `parse_json_string_or_file` anyway (because preset files, other files and literals can be combined).

Reply:
In https://github.com/OCR-D/core/pull/969/commits/db0b84fb7a11d939cf3fdaf00728329ee2efb554 I found a solution by instantiating a throwaway processor and reusing parts of resolve_resource.